### PR TITLE
Add dotnet-stack docs

### PR DESCRIPTION
**Pull request by josalem**
## Summary

Adds the documentation for the new `dotnet-stack` tool.

This PR should not be merged until dotnet/diagnostics#1702 is merged and a subsequent release of the tools is uploaded.

CC @dotnet/dotnet-diag 

### Internal preview

✔️ [Inspect managed stack traces (dotnet-stack)](https://review.docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-stack?branch=pr-en-us-21618)

### CODE REVIEW

File: docs/core/diagnostics/dotnet-stack.md
Comment (sywhang) [inline docs/core/diagnostics/dotnet-stack.md:12]: This part should be changed to match what https://github.com/dotnet/docs/pull/21617 did.

File: docs/core/diagnostics/index.md
Comment (sywhang) [inline docs/core/diagnostics/index.md:52]: There is another place you should try to add which is under `docs\fundamentals\diagnostics\toc.yml` which has links to each tools as a side panel that shows up in https://docs.microsoft.com/en-us/dotnet/core/diagnostics/

File: docs/core/diagnostics/dotnet-stack.md
Comment (sywhang) [inline docs/core/diagnostics/dotnet-stack.md:39]: I think the part about it being a cross-platform tool already makes this part pretty clear so we can leave this out.

File: docs/core/diagnostics/dotnet-stack.md
Comment (sywhang) [inline docs/core/diagnostics/dotnet-stack.md:37]: What do you think about changing this to `Captures and writes out the managed stacks for all threads in the target .NET process?`

File: docs/core/diagnostics/dotnet-stack.md
Comment (sywhang) [inline docs/core/diagnostics/dotnet-stack.md:8]: Wasn't there a runtime bug in 3.0 that prevented this from working so this only works from 3.1, or am I misremembering this?

**Review by sywhang — COMMENTED** [review]
yaaay documentation culture on diagnostics team 💯

File: docs/core/diagnostics/dotnet-stack.md
Comment (josalem) [reply docs/core/diagnostics/dotnet-stack.md:8]: That was gcdump!

Comment (sywhang) [reply docs/core/diagnostics/dotnet-stack.md:8]: you're right :) nvm then!

File: docs/core/diagnostics/dotnet-stack.md
Comment (josalem) [reply docs/core/diagnostics/dotnet-stack.md:37]: That works too.  I used report here since I copied this over from the dotnet-trace docs which seemed to try to match the phrasing with the command verb, e.g., collect for `dotnet-trace collect`.

Comment (sywhang) [reply docs/core/diagnostics/dotnet-stack.md:37]: Interesting. Some tools have more than one commands (i.e. counters, trace both have/will have collect/monitor) so it might not make the most sense to use the commands to describe the tools for those two tools.

Do you anticipate `dotnet-stack` to have more than just `report` command anytime soon?

Comment (josalem) [reply docs/core/diagnostics/dotnet-stack.md:37]: I don't anticipate `dotnet-stack` to have more verbs than `report`.  I already changed it to your suggestion locally.  I just need to push it up.

**Review by sywhang — APPROVED** [review]
👍

File: docs/core/diagnostics/dotnet-stack.md
Comment (adegeo) [inline docs/core/diagnostics/dotnet-stack.md:3]: Improve search and match other diagnostic tool design (pending in PR)

```suggestion
title: dotnet-stack diagnostic tool - .NET CLI
description: Learn how to install and use the dotnet-stack command-line tool which captures and prints the managed stacks for all threads in the target .NET process.
```

File: docs/core/diagnostics/dotnet-stack.md
Comment (adegeo) [inline docs/core/diagnostics/dotnet-stack.md:16]: Fix tool name
```suggestion
  To install the latest release version of the `dotnet-stack` [NuGet package](https://www.nuget.org/packages/dotnet-stack), use the [dotnet tool install](../tools/dotnet-tool-install.md) command:
```

**Review by adegeo — APPROVED** [review]

**Comment by shirhatti** [comment]
Should we add a gotcha to the document that this results in rundown making it unsuitable for use in certain production environments?

**Comment by sywhang** [comment]
I'll follow up with a PR to add known caveats section to all the diagnostic tools once they're published

File: docs/core/diagnostics/dotnet-stack.md
Comment (noahfalk) [inline docs/core/diagnostics/dotnet-stack.md:8]: I think this issue applies to other tools as well so perhaps best for a separate PR. I expect customers to treat "Applies to" as "what .NET runtime version can I diagnose with the tool"
```suggestion
**This article applies to:** ✔️ .NET Core 3.0 and later versions
```

File: docs/core/diagnostics/dotnet-stack.md
Comment (noahfalk) [inline docs/core/diagnostics/dotnet-stack.md:44]:
```suggestion
* Utilizes [`EventPipe`](./eventpipe.md) tracing provided by the .NET Core runtime.
```

File: docs/core/diagnostics/dotnet-stack.md
Comment (noahfalk) [inline docs/core/diagnostics/dotnet-stack.md:65]:
```suggestion
Prints the stack trace for each thread in the target process.
```

File: docs/core/diagnostics/dotnet-stack.md
Comment (noahfalk) [inline docs/core/diagnostics/dotnet-stack.md:71]: This is just an aside and you don't need to change anything. I know dotnet-stack didn't start the trend but I find the name of the "--name" option is super ambiguous : ) Without docs I would never guess what this meant.

**Review by noahfalk — APPROVED** [review]
LGTM (modulo a few nits)

File: docs/core/diagnostics/dotnet-stack.md
Comment (josalem) [reply docs/core/diagnostics/dotnet-stack.md:8]: This tool should work for 3.0 as well.

**Comment by josalem** [comment]
@sywhang & @noahfalk I added a blurb at the end about the output of the report for people who may want to write scripts around the output.

**Comment by josalem** [comment]
The tool is merged into the diagnostics repo.  Should we merge these now or wait till the first release with the new tool?

**Comment by sywhang** [comment]
@josalem wait until the official build has been published -- these docs go live pretty shortly after you merge so it'll be public before the tool is.

**Comment by adegeo** [comment]
Changing to draft mode, this prevents any merge.

**Comment by hoyosjs** [comment]
@sywhang this is ready for merge.

**Review by adegeo — APPROVED** [review]

File: docs/core/diagnostics/dotnet-stack.md
Comment (adegeo) [inline docs/core/diagnostics/dotnet-stack.md:140]: There should be another section here at the end like `## Next steps` which has links to the diagnostics overview and other related content.

File: docs/core/diagnostics/dotnet-stack.md
Comment (adegeo) [inline docs/core/diagnostics/dotnet-stack.md:4]:
```suggestion
ms.date: 04/21/2021
```

**Review by adegeo — DISMISSED** [review]
Looks good, just one minor thing to add which I commented above.

File: docs/core/diagnostics/dotnet-stack.md
Comment (josalem) [reply docs/core/diagnostics/dotnet-stack.md:140]: Added

File: docs/core/diagnostics/dotnet-stack.md
Comment (IEvangelist) [inline docs/core/diagnostics/dotnet-stack.md:142]:
```suggestion
## Next steps
```

File: docs/core/diagnostics/dotnet-stack.md
Comment (IEvangelist) [inline docs/core/diagnostics/dotnet-stack.md:61]:
```suggestion
| Command                                     | Description                                                   |
|---------------------------------------------|---------------------------------------------------------------|
| [dotnet-stack report](#dotnet-stack-report) | Prints the stack trace for each thread in the target process. |
| [dotnet-stack ps](#dotnet-stack-ps)         | Lists the dotnet processes that traces can be collected from. |
```

File: docs/core/diagnostics/dotnet-stack.md
Comment (IEvangelist) [inline docs/core/diagnostics/dotnet-stack.md:130]:
```suggestion
  ```console
```

**Review by IEvangelist — COMMENTED** [review]
A few nits... when these are addressed let's :shipit:

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**
